src/Insightly/Parser/LinkParser.php

```diff
@@ -29,6 +29,18 @@ public static function parseToResult($data)
         $link->setRole(!empty($data['ROLE']) ? $data['ROLE'] : null);
         $link->setDetails(!empty($data['DETAILS']) ? $data['DETAILS'] : null);
 
+        // Work around to take into account the new format of links
+        // This needs a more robust solution, see https://jira.uitdatabank.be/browse/PROJ-156
+        if (isset($data['LINK_OBJECT_NAME'], $data['LINK_OBJECT_ID'])) {
+            if ($data['LINK_OBJECT_NAME'] === 'Contact') {
+                $link->setContactId($data['LINK_OBJECT_ID']);
+            }
+
+            if ($data['LINK_OBJECT_NAME'] === 'Organisation') {
+                $link->setContactId($data['LINK_OBJECT_ID']);
+            }
+        }
+
```

Comment:
How about:

```
$linkObjectName = $data['LINK_OBJECT_NAME'] ?? '';
$linkObjectId = $data['LINK_OBJECT_ID'] ?? null;
if (($linkObjectName === 'Contact' || $linkObjectName === 'Organisation') && !is_null($linkObjectId)) {
    $link->setContactId($linkObjectId);
}
```

?

Reply:
That doesn't really feel like a massive improvement over what @LucWollants suggested imho. Both are a workaround and both are "good enough" I think. The rest is very much preference.

Reply:
Should the second `$link->setContactId($data['LINK_OBJECT_ID']);` be some kind of `$link->setOrganisationId($data['LINK_OBJECT_ID']);` perhaps?

Reply:
If it's intentional `setContactId` twice I feel like a single `if` would better radiate that intent, that it's the same for both

Reply:
Good that you spotted this. This was a copy/paste error!

Reply:
Aha okay! Then 2 ifs make sense of course 😄